docs/core/lightning-network.md

```diff
@@ -0,0 +1,29 @@
+---
+id: lightning-network
+title: Lightning Network Data Types
+---
+
+Bitcoin-S offers support for many different Lightning Network data types.
+Here are some examples:
+
+```scala mdoc:invisible
+import org.bitcoins.core.protocol.ln._
+import org.bitcoins.core.protocol.ln.currency._
+import org.bitcoins.core.protocol.ln.node._
+import org.bitcoins.crypto._
+import scala.util._
+import scodec.bits._
+```
+
+```scala mdoc:compile-only
```

Comment:
So I don't think we want `compile-only` here, I think we want  `to-string` so we get nice comments on the doc that look like this

```
// this generates a random private key
val privkey = ECPrivateKey()
// privkey: ECPrivateKey = Masked(ECPrivateKeyImpl)
val pubkey = privkey.publicKey
// pubkey: org.bitcoins.crypto.ECPublicKey = ECPublicKey(033ee1b9566f60866af0aae5b02cc5148b91a1cf3d3b3f0b35c9f6368b3b3a9012)

val segwitAddress = {
    // see https://bitcoin.org/en/glossary/pubkey-script
    // for reading resources on the details of scriptPubKeys
    // pay-to-witness-pubkey-hash scriptPubKey V0
    val scriptPubKey = P2WPKHWitnessSPKV0(pubkey)
    Bech32Address(scriptPubKey, TestNet3)
}
// segwitAddress: Bech32Address = tb1qaftdnhe2svfu40kg64nf4nv99kjan9phpuaggw

println(segwitAddress.toString)
// tb1qaftdnhe2svfu40kg64nf4nv99kjan9phpuaggw
```

https://bitcoin-s.org/docs/next/core/addresses